### DISCUSSION TITLE
Phalcon 5 can't be installed or used on PHP 8.2

### DISCUSSION THREAD

**Discussion by svdigital-development** [phalcon/cphalcon#16463] [General]
**Describe the bug**
We are UNABLE to use Phalcon 5 with PHP 8.2 and this is preventing us to upgrade our applications. PHP 8.1 will end its active support in one month and it's unbelievable that it's impossible to upgrade to php8.2 using Phalcon.

The documentation is unuseful: https://docs.phalcon.io/5.0/it-it/installation

"pecl install phalcon" on Debian generates a HUGE amount of warnings on Zephir.

After all those warnings, the installation stops and gets stuck. The pointer remains on its position like it's waiting for a process to end, but in reality the server is not doing anything. I tried waiting even half an hour, but nothing happens.

I tried to compile from sources, but I get the same errors.

I can't find any new sury repository for php8.2-phalcon5 (until php 8.1 it was SOOO simple and handy to install Phalcon so easily).

I read #16401 and #15973 but no one explains how to fix this bug/issue.

The documentation doesn't explain how to install Phalcon on PHP 8.2.

**To Reproduce**

> Provide output if related. Provide coredump if any. Use https://docs.phalcon.io/en/latest/generating-backtrace as reference.

Try to install Phalcon5 with php 8.2.

**Expected behavior**

Phalcon5 is correctly installed in a few seconds and available for use on PHP 8.2 (fpm/cli).

**Details**

 - Phalcon version: 5.1/5.2
 - PHP Version: 8.2.7
 - Operating System: Debian 10
 - Installation type: Compiling from source or installing via package manager PECL
 - Zephir version (if any): don't know how to find out
 - Server: Nginx

**Comment by s-ohnishi** [comment]
I also encountered the same problem.
However, it was a combination of PHP8.1 and Phalcon5.4.0.
I was building this PHP8.1 with Docker Desktop for windows (WSL2).
Just to be sure, I changed the memory allocation to WSL2 from 4GB to 8GB and tried again, and the build was successful.
I also tried it with PHP8.2, and the build completed without any problems.
It may require more memory than Phalcon5.3.x.

The Dockerfile is roughly written as follows.
```Dockerfile 
FROM php:8.2-fpm-bullseye
(Installation of necessary packages, etc....omitted)
ENV PHALCON_VERSION="5.4.0"

RUN set -eux; \
   pecl install phalcon-${PHALCON_VERSION}; \
   pecl clear-cache
```

**Comment by Jeckerson** [comment]
You need at least 3Gb of RAM to compile Phalcon, ideally more. Compilation warning are to ignore.

If you are installing via pecl inside VM or in machine with limited RAM resource, add swap file with 5Gb.
Here is example how to do that: https://gist.github.com/Jeckerson/2322849d9f5dbbc9543c99e49341177e

**Comment by Jeckerson** [comment]
https://github.com/phalcon/cphalcon/assets/3289702/cbd5914a-0ce8-44b0-bebc-dc0426bf2acc

Here is example how RAM grows during single thread compilation, which takes ±3 minutes.

**Comment by s-ohnishi** [comment]
I've got it.
We have to be careful with cheap rental servers (as well as machines that don't have a lot of resources).

I don't know what @svdigital-development  environment is, but I hope things go well.

**Comment by Jeckerson** [comment]
@s-ohnishi ideally is to pre-compile ready to use `Dockerfile` with all your stuff in: phalcon, another libs and dependencies and the code itself.

**Comment by s-ohnishi** [comment]
I see, I didn't notice that.
However, even if you create an image for yourself, it is still a "personal" image, so it is not very reliable, and what you include will not necessarily be useful to others.
It would be nice to have an official Phalcon image with only the minimum packages installed.
(I wish it was built on Debian)

**Comment by Jeckerson** [comment]
@s-ohnishi This is on todo list, there are some ready to use images, but still plenty things to do.

https://github.com/phalcon/docker - track this repository for further updates.

**Comment by s-ohnishi** [comment]
@Jeckerson
I didn't know this project existed.
Also the official image of Docker.
Is there anything I can do to help?
However, I am a personal user who has built images locally to some extent, and I am also a novice git user.
Above all, my English proficiency is limited to the point where I can't post without the help of a "translation site".

**Comment by Jeckerson** [comment]
@s-ohnishi Find me in our Discord sever for further ideas - https://phalcon.io/discord

**Comment by kgrammer** [comment]
FYI... I had two occurances of pecl install phalcon doing the same as the original poster... meaning it started installing, showed the zephir warnings and then never completed or returned.  When I eventually aborted the pecl install and ran htop, I noticed that the swap space was maxed out.

I was able to reboot the server and then the pecl install phalcon command completed as expected.

This was on a server with 4gb of memory and a server with 6gb of memory.  In both cases the server reboot "fixed" the issue and allowed pecl to install phalcon.
